test/suite/test_verify.py

```diff
@@ -137,8 +163,67 @@ def test_verify_api_75pct_null(self):
         self.assertRaisesWithMessage(wiredtiger.WiredTigerError,
-            lambda: self.session.verify('table:' + self.tablename, None),
+            lambda: self.session.verify('table:' + self.tablename, "read_corrupt"),
             "/WT_SESSION.verify/")
+        self.assertEqual(self.count_file_contains("stderr.txt",
+            "calculated block checksum doesn't match expected checksum"), 1)
+
+    def test_verify_api_read_corrupt_pages(self):
+        """
+        Test verify via API, on a table that is purposely corrupted in
+        multiple places. A verify operation with read_corrupt on should
+        result in multiple checksum errors being logged.
+        """
+        params = 'key_format=S,value_format=S'
+        self.session.create('table:' + self.tablename, params)
+        self.populate(self.tablename)
+        with self.open_and_position(self.tablename, 25) as f:
+            for i in range(0, 100):
+                f.write(b'\x01\xff\x80')
+        with self.open_and_position(self.tablename, 50) as f:
+            for i in range(0, 100):
+                f.write(b'\x01\xff\x80')
+        with self.open_and_position(self.tablename, 75) as f:
+            for i in range(0, 100):
+                f.write(b'\x01\xff\x80')
+
+    def test_verify_api_corrupt_first_page(self):
+        """
+        Test that verify works when the first child of an internal node is corrupted. A verify
+        operation with read_corrupt on should result in a checksum errors being logged.
+        """
+        params = 'key_format=S,value_format=S'
+        self.session.create('table:' + self.tablename, params)
+        self.populate(self.tablename)
+
+        # wt verify -d dump_address performs a depth-first traversal of the BTree. So the first 
+        # leaf page it prints is the first child of its parent. Grab the offset of this one so we
+        # can corrupt it.
+        self.runWt(['verify', '-d', 'dump_address', 'table:' + self.tablename], outfilename='dump.out')
+
+        # Grab the offset position of the first page.
+        offset = 0
+        lines = open('dump.out').readlines()
+        for line in lines:
+            m = re.search('(\d+)-(\d+).*row-store leaf', line)
+            if m:
+                offset = int((int(m.group(2)) - int(m.group(1)))/2)
+                break
+        
+        # Open the file and corrupt the first page. 
+        with self.open_and_offset(self.tablename, offset) as f:
+            for i in range(0, 100):
+                f.write(b'\x01\xff\x80')
+
+        # open_and_position closed the session/connection, reopen them now.
+        self.conn = self.setUpConnectionOpen(".")
+        self.session = self.setUpSessionOpen(self.conn)
+        self.assertRaisesWithMessage(wiredtiger.WiredTigerError,
+            lambda: self.session.verify('table:' + self.tablename, "read_corrupt"),
+            "/WT_SESSION.verify/")
+
+        self.assertEqual(self.count_file_contains("stderr.txt",
```

Comment:
Shouldn't we be removing this from the PR as it was causing issues?

Reply:
The assertion will cause issues if it is searching for the message `"calculated block checksum of"` as the message was changed to that in WT-10563. Reverting the message back to the older version `"calculated block checksum doesn't match expected checksum"` shouldn't cause any issues.